From 9f066de05b50af9bab8db3a26d75bab4aff52fce Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 2 Feb 2016 19:56:26 +0200
Subject: [PATCH 0776/1240] fix: comphy: usb3: fix reference clock to be on
 group 1 (25Mhz)

- Fix reference clock to be on group 1 - 25Mhz
- Update delay to be 10ms
- Update debug print

Change-Id: I0611b2e28dd209188d33404d52821a03b6d0e7aa
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27153
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
---
 drivers/phy/comphy_cp110.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/phy/comphy_cp110.c b/drivers/phy/comphy_cp110.c
index 3901ca0..a9f22da 100644
--- a/drivers/phy/comphy_cp110.c
+++ b/drivers/phy/comphy_cp110.c
@@ -225,7 +225,7 @@ static int comphy_usb3_power_up(u32 lane, void __iomem *hpipe_base, void __iomem
 	void __iomem *comphy_addr = COMPHY_ADDR(comphy_base, lane);
 
 	debug_enter();
-	debug("stage: RFU configurations- hard reset comphy\n");
+	debug("stage: RFU configurations - hard reset comphy\n");
 	/* RFU configurations - hard reset comphy */
 	mask = COMMON_PHY_CFG1_PWR_UP_MASK;
 	data = 0x1 << COMMON_PHY_CFG1_PWR_UP_OFFSET;
@@ -267,8 +267,8 @@ static int comphy_usb3_power_up(u32 lane, void __iomem *hpipe_base, void __iomem
 	/* Set PLL ready delay for 0x2 */
 	reg_set(hpipe_addr + HPIPE_CLK_SRC_LO_REG,
 		0x2 << HPIPE_CLK_SRC_LO_PLL_RDY_DL_OFFSET, HPIPE_CLK_SRC_LO_PLL_RDY_DL_MASK);
-	/* Set reference clock to come from group 2 */
-	reg_set(hpipe_addr + HPIPE_MISC_REG, 0x1 << HPIPE_MISC_REFCLK_SEL_OFFSET, HPIPE_MISC_REFCLK_SEL_MASK);
+	/* Set reference clock to come from group 1 - 25Mhz */
+	reg_set(hpipe_addr + HPIPE_MISC_REG, 0x0 << HPIPE_MISC_REFCLK_SEL_OFFSET, HPIPE_MISC_REFCLK_SEL_MASK);
 	/* Set reference frequcency select - 0x2 */
 	mask = HPIPE_PWR_PLL_REF_FREQ_MASK;
 	data = 0x2 << HPIPE_PWR_PLL_REF_FREQ_OFFSET;
@@ -306,7 +306,7 @@ static int comphy_usb3_power_up(u32 lane, void __iomem *hpipe_base, void __iomem
 		0x0 << HPIPE_RST_CLK_CTRL_PIPE_RST_OFFSET, HPIPE_RST_CLK_CTRL_PIPE_RST_MASK);
 
 	/* wait 5ms - for comphy calibration done */
-	mdelay(5);
+	mdelay(10);
 
 	debug("stage: Check PLL\n");
 	/* Read lane status */
-- 
1.9.1

